src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -424,3753 +416,3744 @@ private void BuildApplicationInitializerBody(IndentedStringBuilder writer, XamlO
+						{
+							if (member.Member.Name == "_UnknownContent") // So : FindType(member.Owner.Type) is INamedTypeSymbol type && IsCollectionOrListType(type)
+							{
+								foreach (var item in member.Objects)
+								{
+									writer.AppendLineInvariant($"{closureName}.Add(");
+									using (writer.Indent())
+									{
+										BuildChild(writer, member, item);
+									}
+									writer.AppendLineInvariant(");");
+								}
+							}
+							else if (!IsType(objectDefinition.Type, member.Member.DeclaringType))
+							{
+								var ownerType = GetType(member.Member.DeclaringType);
+
+								var propertyType = GetPropertyType(member.Member);
+
+								if (IsExactlyCollectionOrListType(propertyType))
+								{
+									// If the property is specifically an IList or an ICollection
+									// we can use C#'s collection initializer.
+									writer.AppendLineInvariant(
+										"{0}.Set{1}({2}, ",
+										GetGlobalizedTypeName(ownerType.ToDisplayString()),
+										member.Member.Name,
+										closureName
+									);
+
+									using (writer.BlockInvariant("new[]"))
+									{
+										foreach (var inner in member.Objects)
+										{
+											BuildChild(writer, member, inner);
+											writer.AppendLine(",");
+										}
+									}
+
+									writer.AppendLineInvariant(");");
+								}
+								else if (IsCollectionOrListType(propertyType))
+								{
+									// If the property is a concrete type that implements an IList or
+									// an ICollection, we must get the property and call add explicitly
+									// on it.
+									var localCollectionName = $"{closureName}_collection_{_collectionIndex++}";
+
+									writer.AppendLineInvariant(
+										$"var {localCollectionName} = {ownerType.ToDisplayString()}.Get{member.Member.Name}({closureName});"
+									);
+
+									foreach (var inner in member.Objects)
+									{
+										writer.AppendLine($"{localCollectionName}.Add(");
+
+										BuildChild(writer, member, inner);
+
+										writer.AppendLineInvariant(");");
+									}
+								}
+								else
+								{
+									// If the property is specifically an IList or an ICollection
+									// we can use C#'s collection initializer.
+									writer.AppendLineInvariant(
+										"{0}.Set{1}({2}, ",
+										GetGlobalizedTypeName(ownerType.ToDisplayString()),
+										member.Member.Name,
+										closureName
+									);
+
+									if (member.Objects.Count() == 1)
+									{
+										BuildChild(writer, member, member.Objects.First());
+									}
+									else
+									{
+										throw new InvalidOperationException($"The property {member.Member.Name} of type {propertyType} does not support adding multiple objects.");
+									}
+
+									writer.AppendLineInvariant(");");
+								}
+							}
+							else
+							{
+								// GenerateWarning(writer, $"Unknown type {objectDefinition.Type} for property {member.Member.DeclaringType}");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5305744)